source/administration/object-management/data-compression.rst

```diff
@@ -0,0 +1,191 @@
+.. _minio-data-compression:
+
+================
+Data Compression
+================
+
+.. default-domain:: minio
+
+.. contents:: Table of Contents
+   :local:
+   :depth: 2
+
+
+Overview
+--------
+
+MinIO Server supports compressing objects to reduce disk usage.
+Objects are compressed before they are written to disk and uncompressed when read, making the compresson process transparent to client applications and services.
+
+Depending on the type of data, compression may also increase overall throughput.
+Write throughput for compressible data is generally 500MB per second or greater per available CPU.
+Decompression is typically at least 1 GB per second.
+
+Data compression is a global option, configured settings apply to all buckets in a deployment.
+You can control which objects are compressed by specifying the desired file extensions and media types.
+
+.. admonition:: Existing objects are not modified
+   :class: note
+
+   Enabling, disabling, or updating a deployment's compression settings does not modify existing objects.
+   New objects are compressed according to the settings in effect at the time they are created.
+
+
+Incompressible Data
+~~~~~~~~~~~~~~~~~~~
+
+Some types of objects cannot be effectively compressed, such as video or already compressed data.
+MinIO does not compress common incompressible file types, even if they are specified in the compression configuration.
+
+Objects of these types are never compressed:
+
+File extensions
+
+- ``gz``, ``bz2``, ``rar``, ``zip``, ``7z``, ``xz``, ``mp4``, ``mkv``, ``mov``
+
+Media types
+- ``video/*``, ``audio/*``, ``application/zip``, ``application/x-gzip``, ``application/x-bz2``, ``application/x-compress``, ``application/x-xz``
+
+	
+Data Compression and Encryption
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+MinIO supports encrypting compressed objects but recommends against encryption for most deployments.
+An important concern is a 3rd party may be able to learn details of an object's contents by comparing the uncompressed and compressed file sizes.
```

Comment:
Perhaps link to the relevant readme for more details. How much detail about the security considerations is appropriate here?

Reply:
I *think* it's sufficient to link to Andrea's blog post and let the user decide from there. We could drop all of line 54 and integrate into line 53 as

```
...recommends `against combining encryption and compression for most deployents <bloglink>`__
```

Reply:
[This is the blog post in question](https://blog.min.io/c-e-compression-encryption/)

Tons of good info, but old enough that it says we don't have an option to encrypt compressed objects. (Feature added January 2021.)

Reply:
Linked to newer blog post from Klaus, which also discusses encrypting compressed objects.